cpp/autosar/test/rules/M9-3-3/test.cpp

```diff
@@ -193,3 +193,24 @@ void test_template() {
 class Z3 {
   void f(int) = delete; // COMPLIANT
 };
+
+class Z4 {
+public:
+  int values[128];
+  template <typename T>
+  void fill(const T &val) { // COMPLIANT[FALSE_NEGATIVE|TRUE_NEGATIVE] -
+                            // exception not specified in the
+    // standard, we opt to not raise an issue because the template can be both
+    // compliant and non-compliant depending on instantiations.
+    for (auto &elem : values) {
+      elem = val;
+    }
+  }
+  constexpr int &front() noexcept { return values[0]; } // COMPLIANT
+};
+
+void fp_reported_in_381() {
+  Z4 z;
+  int i = z.front();
```

Comment:
Missing compliance annotation.

Reply:
the rule alerts on functions, so their declarations not the usage (the compliance annotation is on [line 209 ](https://github.com/knewbury01/codeql-coding-standards/blob/knewbury01/fix-381/cpp/autosar/test/rules/M9-3-3/test.cpp#L209) but the usage snip is probably in there as I was checking if template instantiation was a factor of this fp/fn case...

good to check tho! thanks!